### ISSUE TITLE
Parallelize artifact building

### ISSUE THREAD

**Issue by kzantow** [anchore/syft#1219]
Currently, the build process creates snapshot artifacts for all the supported platforms. This step is the slowest of the entire process ([18 minutes recently!](https://github.com/anchore/syft/actions/runs/3085306890/jobs/4988444941#step:8:1)) and is running for each PR. The `goreleaser` command is called with [`parallelism = 1`](https://github.com/anchore/syft/blob/ad263e656222dd15034984725d8513f2bad1e14c/Makefile#L367) which is likely related to macOS notarization, which doesn't even run until a release. We could split this up so everything _except_ macOS runs in parallel, and macOS runs separately or possibly run everything in parallel for PRs, but fall back to this sequential build for releases.

It is unclear how much this would speed up the build, but it could be a considerable amount.

**Comment by kzantow** [comment]
The default `goreleaser` behavior is already to run in parallel with the number of cores present. Some testing indicated there is no better setting we can find here to run all the builds.
